File: algo_8_math/L326.cpp
//
// Created by LiHeng on 2020/6/30.
//
#include <iostream>
using namespace std;

class Solution {
public:
    bool isPowerOfThree(int n) {
        if(n<1)
            return false;
        while(n%3 == 0)
            n = n/3;
        return n==1;
    }
};

File: python-algorithm/algo_09_math/L326.py
# -*- coding: utf-8 -*-


class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n == 0:
            return False
        while n % 3 == 0:
            n = n//3
        return n == 1

# 进阶：不用循环/递归 该怎么做?